### PR TITLE
feat: support SDK auth in AzureSD

### PR DESCRIPTION
**Pull request by slashpai**
feat: support SDK auth in AzureSD

Related-to #6584

## Description

_Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue._

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [x] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Verification

Please check the [Prometheus-Operator testing guidelines](../TESTING.md) for recommendations about automated tests.

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
feat: support SDK auth in AzureSD
```

### CODE REVIEW

File: pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go [hunk 1]
```diff
@@ -518,7 +518,9 @@ type AzureSDConfig struct {
 	Environment *string `json:"environment,omitempty"`
 	// # The authentication method, either OAuth or ManagedIdentity.
```
Comment (simonpasquier) [inline pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go:519]:
```suggestion
	// # The authentication method, either `OAuth`, `ManagedIdentity` or `SDK`.
```

File: pkg/prometheus/resource_selector.go [hunk 2]
```diff
@@ -829,7 +829,7 @@ func (rs *ResourceSelector) SelectScrapeConfigs(ctx context.Context, listFn List
 			continue
 		}
 
-		if err = rs.validateAzureSDConfigs(ctx, sc); err != nil {
+		if err = rs.validateAzureSDConfigs(ctx, cpf.Version, sc); err != nil {
```
Comment (simonpasquier) [inline pkg/prometheus/resource_selector.go:832]: rs already knows about the monitoringv1.PrometheusInterface object so no need to pass the version.

File: pkg/prometheus/resource_selector.go [hunk 3]
```diff
@@ -1033,10 +1033,23 @@ func (rs *ResourceSelector) validateEC2SDConfigs(ctx context.Context, sc *monito
...
+	version, err := semver.ParseTolerant(promVersion)
+	if err != nil {
+		return fmt.Errorf("failed to parse Prometheus version: %w", err)
+	}
+
+	minimumVersionSDKAuth := version.GTE(semver.MustParse("2.52.0"))
+
 	for i, config := range sc.Spec.AzureSDConfigs {
+		authMethod := ptr.Deref(config.AuthenticationMethod, "")
+		if authMethod == "SDK" && !minimumVersionSDKAuth {
```
Comment (simonpasquier) [inline pkg/prometheus/resource_selector.go:1047]: (nit)
```suggestion
		if authMethod == "SDK" && version.LT(semver.MustParse("2.52.0")) {
```

**Comment by slashpai** [comment]
@simonpasquier addressed comments PTAL

**Review by simonpasquier — APPROVED** [review]

**Pull request merged**